apps/editors/tests/test_views.py

```diff
@@ -668,22 +668,29 @@ def addon_file(self, *args, **kw):
         a = create_addon_file(*args, listed=self.listed, **kw)
         name = args[0]  # Add-on name.
         self.addons[name] = a['addon']
-        # If this is an add-on we expect to be in the queue, then add it.
-        if name in getattr(self, 'expected_names', []):
-            self.expected_addons.append(a['addon'])
         return a['addon']
 
     def get_queue(self, addon):
         version = addon.latest_version.reload()
         eq_(version.current_queue.objects.filter(id=addon.id).count(), 1)
 
+    def get_expected_addons_by_names(self, names):
+        expected_addons = []
+        self.addons = {}
+        files = self.generate_files()
+        for addon in files:
+            for name in sorted(names):
+                if name == addon:
+                    expected_addons.append(files[addon])
```

Comment:
I think this could be rewritten to be a little less repetitive as:
````
for name in sorted(names):
    if name in files:
        expected_addons.append(files[addon])
````